packages/block-library/src/image/edit.js

```diff
@@ -152,6 +153,16 @@ class ImageEdit extends Component {
 				captionFocused: false,
 			} );
 		}
+
+		if ( this.state.autoFocus && this.container ) {
+			this.setState( {
```

Comment:
Rather than setting this in state, could the focusing just occur as part of the logic for `onSelectImage` ? 

The issue with this is that `setState` will force a render, and is generally discouraged to be incorporated into `componentDidUpdate`, since the nature of the lifecycle method is such that it means we'll have a minimum of two renders (when ideally it'd just be one max per "change") and leaves open the potential for an infinite loop (since this will re-call `componentDidUpdate`).

Reply:
The problem here is that the DOM elements do not yet exist when onSelectImage gets called. The `if` statement is intended to stop the infinite loop.

Reply:
Another option could be to assign this value as an instance property, so that we don't need to call `setState` here. It's a bit more fragile / non-standard, but specifically avoids the unneeded render.

Effectively, the difference would be:

```js
this.setState( {
	isEditing: false,
} );
this.autoFocusCaption = true;
```

```js
if ( this.autoFocusCaption && this.myRef.current ) {
	this.autoFocusCaption = false;
	// Or: 
	// delete this.autoFocusCaption;
	const caption = this.myRef.current.querySelector( 'figcaption' );
	// ...
```